compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -38,13 +38,13 @@ object ProtoTypes {
     def isCompatible(tp: Type, pt: Type)(using Context): Boolean =
       (tp.widenExpr relaxed_<:< pt.widenExpr) || viewExists(tp, pt)
 
-    /** Like isCompatibe, but using a subtype comparison with necessary eithers
-     *  that don't unnecessarily truncate the constraint space, returning false instead.
+    /** Like normalize and then isCompatible, but using a subtype comparison with
+     *  necessary eithers that does not unnecessarily truncate the constraint space,
+     *  returning false instead.
      */
     def necessarilyCompatible(tp: Type, pt: Type)(using Context): Boolean =
-      val tpw = tp.widenExpr
-      val ptw = pt.widenExpr
-      necessarySubType(tpw, ptw) || tpw.isValueSubType(ptw) || viewExists(tp, pt)
+      val tpn = normalize(tp, pt, followIFT = !defn.isContextFunctionType(pt))
+      necessarySubType(tpn, pt) || tpn.isValueSubType(pt) || viewExists(tpn, pt)
```

Comment:
Sorry for so many questions on such a simple PR. But I noticed that you don't widen `pt` here anymore. Can't this change the result of comparing the types for other examples?

Reply:
In theory it could, but in practice it did not. Conceptually, the correct thing is not to widen here, so unless we see an example where it causes a problem that should be the default.